src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +63,16 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    /**
+     * Return a boolean value which indicates whether this command is changing the data.
+     *
+     * @throws UnsupportedOperationException this method should be implemented by children classes.
+     */
+
+    public boolean isMutating() {
```

Comment:
You can see this case is similar to abstract LO in both format and our intention. Could you find an alternative solution to implement this method in parent class based on this observation?